controllers/mirrorpeer_controller.go

```diff
@@ -138,6 +162,19 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			logger.Info("Finalizer not found on MirrorPeer. Adding Finalizer")
 			mirrorPeer.Finalizers = append(mirrorPeer.Finalizers, mirrorPeerFinalizer)
 		}
+		if mirrorPeer.Status.Phase == "" || mirrorPeer.Status.Phase == multiclusterv1alpha1.IncompatibleVersion {
+			if mirrorPeer.Spec.Type == multiclusterv1alpha1.Async {
+				mirrorPeer.Status.Phase = multiclusterv1alpha1.ExchangingSecret
+			} else {
+				mirrorPeer.Status.Phase = multiclusterv1alpha1.S3ProfileSyncing
+			}
+			statusErr := r.Client.Status().Update(ctx, &mirrorPeer)
+			if statusErr != nil {
+				logger.Error("Error occurred while updating the status of mirrorpeer. Re-queing request.", "Error ", statusErr)
```

Comment:
```suggestion
				logger.Error("Error occurred while updating the status of mirrorpeer. Requeing request.", "Error ", statusErr)
```
Just as above

Reply:
I thought "Requeing" is not a word. TIL.

Reply:
We are using it above above `Requeuing`. Just a nit, you can change if you want